src/Quantum/MP.jl

```diff
@@ -110,6 +111,17 @@ end
 const MPS = MatrixProduct{State}
 const MPO = MatrixProduct{Operator}
 
+tensors(ψ::TensorNetwork{MatrixProduct{P,Infinite}}, args...) where {P<:Plug} =
+    throw(MethodError("You need to specify the site for an infinite MatrixProduct$P"))
+
+tensors(ψ::TensorNetwork{MatrixProduct{P,Infinite}}, site::Int, args...) where {P<:Plug} =
+    tensors(plug(ψ), ψ, mod1(site, length(ψ.tensors)), args...)
+
+Base.show(io::IO, ψ::TensorNetwork{MatrixProduct{P,Infinite}}) where {P<:Plug} =
+    print(io, "$(typeof(ψ))(#tensors=∞, #labels=∞)")
+
```

Comment:
I think this method shouldn't be overridden, but just the `length` method specialized. Maybe the number of labels it's difficult to obtain?
```suggestion
```

Reply:
Yes, that's it. Since `labels=length(tn.indices)` it was difficult to fix that without overwritting it. However, I think we can change `labels=length(labels(tn))` and this should be fixed.